src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,289 @@
+    except ValueError:
+        # Pandas relies on openpyxl to parse XLSX files.
+        # A strange behaviour of openpyxl prevents pandas from opening files with some formatting properties
+        # (unclear which formatting properties exactly).
+        # Apparently, the excel2json test files have one of the unsupported formatting properties.
+        # Credits: https://stackoverflow.com/a/70537454/14414188
+        with mock.patch("openpyxl.styles.fonts.Font.family.max", new=100):
+            read_df = pd.read_excel(excel_filename)
+    read_df = clean_data_frame(unclean_df=read_df)
+    return read_df
+
+
+def clean_data_frame(unclean_df: pd.DataFrame) -> pd.DataFrame:
+    """
+    This function takes a pd.DataFrame and removes:
+        - Leading and trailing spaces from the column names
+        - Leading and trailing spaces from each cell
+        - Any characters in the cells that are not part of any known language, for example, linebreaks and replaces it
+          with a pd.NA.
+        - Removes all rows that are empty in all columns
+
+    Args:
+        unclean_df: The pd.DataFrame that is to be cleaned
+
+    Returns:
+        pd.DataFrame which has the above-mentioned removed
+    """
+    # Remove leading and trailing blanks in column names and make them lower case
+    cleaned_df = unclean_df.rename(columns=lambda x: x.strip().lower())
+    # Remove the values of all cells that do not at least contain one character of any known language
+    cleaned_df = cleaned_df.applymap(
+        lambda x: str(x).strip() if pd.notna(x) and regex.search(r"[\w\p{L}]", str(x), flags=regex.U) else pd.NA
+    )
+    # drop all the rows that are entirely empty
+    cleaned_df.dropna(axis=0, how="all", inplace=True)
+    return cleaned_df
+
+
+def check_required_columns_raise_error(check_df: pd.DataFrame, required_columns: set[str]) -> None:
+    """
+    This function takes a pd.DataFrame and a set of required column names. It checks if all the columns from the set
+    are in the pd.DataFrame. Additional columns to the ones in the set are allowed. It raises an error if any columns
+    are missing.
+
+    Args:
+        check_df: pd.DataFrame that is checked
+        required_columns: set of column names
+
+    Raises:
+        UserError: if there are required columns missing
+    """
+    # This checks if the required columns are in the Dataframe. Other columns are also permitted.
+    if not required_columns.issubset(set(check_df.columns)):
+        raise UserError(
+            f"The following columns are missing in the excel: " f"{required_columns.difference(set(check_df.columns))}"
+        )
+
+
+def check_duplicate_raise_error(check_df: pd.DataFrame, duplicate_column: str) -> None:
+    """
+    This function checks if a specified column contains duplicate values. Empty cells (pd.NA) also count as duplicates.
+    If there are any duplicate values, it creates a string with the duplicates which are displayed in the error message.
```

Comment:
Does it make sense to raise an error if 2 cells in a column are both pd.NA? It seems counter-intuitive to me.

Reply:
There is not built-in feature in this function that let's us ignore nan. To solve this it requires some extra steps. But in this case there should not be any nan values in the column anyway (that is checked elsewhere) which is why I decided not to take that into account. I mentioned it in the function so that for other application we are aware of it. In that case we could still create another function that takes it into account.